File: scan.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'socket'

IP = '192.168.0.200'.freeze
TIMEOUT = 0.75

(1..128).map do |port|
  Thread.new do
    s = Socket.tcp(IP, port, connect_timeout: TIMEOUT)
    puts port
  rescue Errno::ETIMEDOUT
  else
    s.close
  end
end.each(&:join)

File: scan.py
#!/usr/bin/env python3

from socket import socket, setdefaulttimeout
from threading import Thread

ip = '192.168.0.200'

def check(addr):
    with socket() as s:
        if s.connect_ex(addr) == 0:
            print(addr)

setdefaulttimeout(0.75)

pool = []

for port in range(1, 128):
    t = Thread(target=check, args=[(ip, port)])
    pool.append(t)

[t.start() for t in pool]
[t.join() for t in pool]
